ui/app/pages/confirm-add-suggested-token/confirm-add-suggested-token.component.js

```diff
@@ -118,4 +131,32 @@ export default class ConfirmAddSuggestedToken extends Component {
       </div>
     )
   }
+
+  checkTokenDuplicates(pendingTokens, tokens) {
+    const pending = Object.keys(pendingTokens)
+    const existing = tokens.map(token => token.address)
+    const dupes = pending.filter((proposed) => {
+      return existing.includes(proposed);
+    })
+
+    return dupes.length > 0;
+  }
+
+  /**
+   * Returns true if any pendingTokens both:
+   * - Share a symbol with an existing `tokens` member.
+   * - Does not share an address with that same `tokens` member.
+   * This should be flagged as possibly deceptive or confusing.
+   */
+  checkNameReuse(pendingTokens, tokens) {
+    const duplicates = Object.keys(pendingTokens)
+      .map((addr) => pendingTokens[addr])
+      .filter((token) => {
+        const dupes = tokens.filter(old => old.symbol === token.symbol)
+          .filter(old => old.address !== token.address);
+        return dupes.length > 0;
```

Comment:
There are a bunch of lint failures here - semicolons are disallowed for this repo, for example.

Reply:
Thanks for the ping on this, linted.